lib/src/places_autocomplete_field.dart

```diff
@@ -149,6 +149,9 @@ class _LocationAutocompleteFieldState extends State<PlacesAutocompleteField> {
   @override
   void didUpdateWidget(PlacesAutocompleteField oldWidget) {
     super.didUpdateWidget(oldWidget);
+    if (widget.controller != null) {
+      widget.controller.text = oldWidget.controller.text;
```

Comment:
When I debugged this. I found that widget.controller.text had the previous value -- empty string was the previous value in the code I provided. Seems like this is due to how new widgets are created and their memory is allocated.

Reply:
good one